io/src/main/java/com/github/jorgecastillo/kotlinandroid/io/algebras/persistence/DataSource.kt

```diff
@@ -49,24 +50,26 @@ object DataSource {
     val monadError = IO.monadError()
     return monadError.binding {
       val query = buildFetchHeroesQuery()
-      runInAsyncContext(
+      val result = runInAsyncContext(
           f = { fetchHeroes(query) },
           onError = { monadError.raiseError<List<CharacterDto>>(it) },
           onSuccess = { monadError.pure(it) },
           AC = IO.async()
       ).bind()
-    }.ev()
+      result.bind()
```

Comment:
Double bind?

Reply:
Yes, monad comprehensions were changed in https://github.com/arrow-kt/arrow/pull/630 and now they accept any return value which is then lifted to the monad to be cosnsistent with the way `await/async` works per @pakoito work and suggestion in this area. In this case you can't return an `IO<A>` value as we did before because it would just double wrap it which is what it was doing.